package/gluon-core/files/lib/netifd/proto/gluon_wired.sh

```diff
@@ -7,15 +7,29 @@ init_proto "$@"
 proto_gluon_wired_init_config() {
         proto_config_add_boolean transitive
         proto_config_add_int index
+        proto_config_add_string vxpeer6addr
+        proto_config_add_boolean is_tunnel
 }
 
 xor2() {
         echo -n "${1:0:1}"
         echo -n "${1:1:1}" | tr '0123456789abcdef' '23016745ab89efcd'
 }
 
+is_layer3_device () {
+        local addrlen="$(cat "/sys/class/net/$1/addr_len")"
+        test "$addrlen" -eq 0
```

Comment:
Only L2 devices have MAC addresses, hence an `addr_len` of 0 bytes indicates a L3 device.

```
What:		/sys/class/net/<iface>/addr_len
Date:		April 2005
KernelVersion:	2.6.12
Contact:	netdev@vger.kernel.org
Description:
		Indicates the hardware address size in bytes.
		Values vary based on the lower-level protocol used by the
		interface (Ethernet, FDDI, ATM, IEEE 802.15.4...). See
		include/uapi/linux/if_*.h for actual values.
```

Reply:
That's correct.